packages/cli/src/ui/App.tsx

```diff
@@ -135,6 +135,12 @@ const App = ({ config, settings, startupWarnings = [], version }: AppProps) => {
     consolePatcher.patch();
     registerCleanup(consolePatcher.cleanup);
   }, [handleNewMessage, config]);
+  const numRenders = useRef(0);
+  useEffect(() => {
```

Comment:
lets move this logic out of footer and avoid prop drilling. instead create a DebugProfiler component that is only added to footer if getDebugMode() is true. That component that then track numRenders and handle rendering it to the footer all in one place.

App.tsx is quite big and complex so anything we can do to keep it from growing bigger and more complex is worth it.

Reply:
Done